### DISCUSSION TITLE
Text to SQL chatbot using Langchain's create_sql_query_chain

### DISCUSSION THREAD

**Discussion by Rohannain** [community/community#134920] [Programming Help]
### Body

I am trying to build a Natural Language to SQL chatbot, currently I am using Langchain's  "create_sql_query_chain" to generate the query as follows:

    `chain = create_sql_query_chain(llm, db)'

How do I give custom prompt to the chain to tell the schema of my table. 

One way I have seen in the langchain documentation(reference below):
#https://api.python.langchain.com/en/latest/chains/langchain.chains.sql_database.query.create_sql_query_chain.html
Above they have used PromptTemplate from langchain_core.prompts, but I am not sure how to pass it to my chain any help or lead will be greatly appreciated. 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by geniuszly** [comment]
To customize the prompt in Langchain's `create_sql_query_chain`, you can use the PromptTemplate to define how you want to pass the schema of your table. to begin with, Import the necessary modules:
```python
from langchain import create_sql_query_chain
from langchain_core.prompts import PromptTemplate
```
Then define your prompt template:
```python
prompt_template = PromptTemplate(
    input_variables=["input", "schema"],
    template="Given the following schema:\n{schema}\nGenerate a SQL query for: {input}"
)
```
And pass the prompt to the chain:
```python
chain = create_sql_query_chain(llm, db, prompt=prompt_template)
```
In this setup, `input` would be the natural language question, and `schema` would be the description of your table's schema. When you use this chain, it will automatically incorporate the schema into the prompt before generating the SQL query. This approach allows you to better control how the model understands the database structure and forms the corresponding SQL query.

**Comment by balajikrishna1798** [comment]
I know but i willnot tell.
